assets/src/edit-story/components/form/switch.js

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { rgba } from 'polished';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+
+const SwitchContainer = styled.div`
+  appearance: none;
+  position: relative;
+  background: ${({ theme }) => rgba(theme.colors.fg.v0, 0.3)};
+  border-radius: 100px;
+  color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.86)};
+  font-family: ${({ theme }) => theme.fonts.body2.family};
+  font-size: ${({ theme }) => theme.fonts.body2.size};
+  line-height: ${({ theme }) => theme.fonts.body2.lineHeight};
+  letter-spacing: ${({ theme }) => theme.fonts.body2.letterSpacing};
+  padding: 6px 3px;
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  border: none;
+
+  &:focus {
+    box-shadow: 0 0 0 1px ${({ theme }) => theme.colors.bg.v4};
+  }
+`;
+
+const RadioButton = styled.input.attrs(({ checked, value, id }) => ({
+  type: 'radio',
+  name: 'switch',
+  checked,
+  value,
+  id,
+}))`
+  overflow: hidden;
+  min-width: unset !important;
+  width: 0 !important;
+  height: 0 !important;
+  border: 0 !important;
+  padding: 0 !important;
+  margin: -1px !important;
+  position: absolute;
+
+  &:focus {
+    box-shadow: none !important;
+  }
+`;
+
+const Label = styled.label`
+  z-index: 1;
+  width: 50%;
+  text-align: center;
+  opacity: 0.86;
+
+  ${({ disabled }) =>
+    disabled &&
+    `
+    cursor: default;
+    opacity: 0.3;
+	`}
+`;
+
+const SwitchSpan = styled.span`
+  position: absolute;
+  z-index: 0;
+  top: 3px;
+  left: 3px;
+  display: block;
+  width: calc(50% - 3px);
+  height: 26px;
+  border-radius: 100px;
+  background-color: ${({ theme }) => theme.colors.bg.v8};
+  transition: left 0.15s ease-out;
+
+  ${RadioButton}[value=on]:checked + ${Label} ~ & {
+    left: 50%;
+  }
+`;
+
+function Switch({ value, disabled, onChange, onLabel, offLabel }) {
+  const [flag, setFlag] = useState(value);
```

Comment:
But now it won't update from outside changes. If `value` changes outside of this component, the state won't update. That requires an additional `useEffect`.

As stated, I don't know if we need it to be internally controlled, as it's extra work, but I think it's best practice.

Reply:
Yes, it only accepts the initial props value. But do we need another update after first sync with flag and value? Actually there is no way to update the status outside of this component? What's your thought @barklund

Reply:
Outside updates can be many things. For instance changing the active element on the stage by clicking a different image. It will render the same panels with the same components, but some flag might be false for one image and true for another. If that's the case, this component might not correctly re-render.

Reply:
Remember that this is supposed to be a generic component - there might be future cases where flags can update in other ways.

Reply:
Makes sense for me! 👍

Reply:
Internal state will be updated based on props update outside.